src/Algebra/Construct/Flip/Op.agda

```diff
@@ -0,0 +1,241 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Flipping the arguments of a binary operation preserves many of its
+-- algebraic properties.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Construct.Flip.Op where
+
+open import Algebra
+open import Data.Product using (swap)
+import Data.Sum as Sum
+open import Function.Base using (flip)
+open import Level using (Level)
+open import Relation.Binary.Core using (Rel; _Preserves₂_⟶_⟶_)
+open import Relation.Binary.Definitions using (Symmetric)
+
+private
+  variable
+    a b p ℓ ℓ₁ ℓ₂ : Level
+    A B : Set a
+    ∼ ≈ ≋ : Rel A ℓ
+    ε  : A
+    ⁻¹ : Op₁ A
+    ∙  : Op₂ A
+
+------------------------------------------------------------------------
+-- Properties
+
+preserves₂ : (∼ ≈ ≋ : Rel A ℓ) →
+             ∙ Preserves₂ ∼ ⟶ ≈ ⟶ ≋ → (flip ∙) Preserves₂ ≈ ⟶ ∼ ⟶ ≋
+preserves₂ _ _ _ pres = flip pres
+
+module _ (≈ : Rel A ℓ) (∙ : Op₂ A) where
+
+  associative : Symmetric ≈ → Associative ≈ ∙ → Associative ≈ (flip ∙)
+  associative sym assoc x y z = sym (assoc z y x)
+
+  identity : Identity ≈ ε ∙ → Identity ≈ ε (flip ∙)
+  identity id = swap id
+
+  commutative : Commutative ≈ ∙ → Commutative ≈ (flip ∙)
+  commutative comm = flip comm
+
+  selective : Selective ≈ ∙ → Selective ≈ (flip ∙)
+  selective sel x y = Sum.swap (sel y x)
+
+  idempotent : Idempotent ≈ ∙ → Idempotent ≈ (flip ∙)
+  idempotent idem = idem
+
+  inverse : Inverse ≈ ε ⁻¹ ∙ → Inverse ≈ ε ⁻¹ (flip ∙)
+  inverse inv = swap inv
+
+------------------------------------------------------------------------
+-- Structures
+
+module _ {≈ : Rel A ℓ} {∙ : Op₂ A} where
```

Comment:
With the variables, we don't need this anonymous module right?

Reply:
The variables help with the occurrences of `≈` and `∙` in signatures, but not in the actual definitions (where they have to be passed explicitly, otherwise the unifier is not happy). For example, the line
https://github.com/agda/agda-stdlib/blob/a123013ec41c8341108058764e6c9eaa3d136697/src/Algebra/Construct/Flip/Op.agda#L69
would have to be changed to
```
isSelectiveMagma {≈ = ≈} {∙ = ∙} m = record
```
and likewise for all the other `isXYZStructure` definitions. I find the latter less readable and prefer the module.

Reply:
Good point. Agreed!